lib/matplotlib/colorbar.py

```diff
@@ -495,6 +495,33 @@ def __init__(self, ax, mappable=None, *, cmap=None,
         if isinstance(mappable, contour.ContourSet) and not mappable.filled:
             self.add_lines(mappable)
 
+        # Link the Axes and Colorbar for interactive use
+        self.ax._colorbar = self
+        # Don't navigate on any of these types of mappables
+        if (isinstance(self.norm, (colors.BoundaryNorm, colors.NoNorm)) or
+                isinstance(self.mappable, contour.ContourSet)):
+            self.ax.set_navigate(False)
+
+        # These are the functions that set up interactivity on this colorbar
+        interactive_funcs = ["_get_view", "_set_view", "_set_view_from_bbox",
+                             "drag_pan", "start_pan", "end_pan"]
+        # Store the axes' original interactive functions, so that if cla() is
+        # called we can restore the originals
+        self._orig_interactive_funcs = {"cla": self.ax.cla}
+        for x in interactive_funcs:
+            self._orig_interactive_funcs[x] = getattr(self.ax, x)
+            setattr(self.ax, x, getattr(self, x))
+        # Set the cla function to the cbar's method to override it
+        self.ax.cla = self._cbar_cla
+
+    def _cbar_cla(self):
+        """Function to clear the interactive colorbar state."""
+        for x in self._orig_interactive_funcs:
+            setattr(self.ax, x, self._orig_interactive_funcs[x])
```

Comment:
I think you can just do `delattr` to reset back to the original class method.

Reply:
Yes, that was much simpler. Thanks!